components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/storage/model/EntityTypeEntity.java

```diff
@@ -0,0 +1,134 @@
+package com.sap.cloud.cmp.ord.service.storage.model;
+
+import com.sap.olingo.jpa.metadata.core.edm.annotation.EdmIgnore;
+import com.sap.olingo.jpa.metadata.core.edm.annotation.EdmProtectedBy;
+import org.eclipse.persistence.annotations.Convert;
+import org.eclipse.persistence.annotations.TypeConverter;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+@Entity(name = "entityType")
+@Table(name = "tenants_entity_types")
+public class EntityTypeEntity {
+    @javax.persistence.Id
+    @Column(name = "id")
+    @Convert("uuidConverter")
+    @TypeConverter(name = "uuidConverter", dataType = Object.class, objectType = UUID.class)
+    private UUID Id;
+
+    @Column(name = "ord_id", length = 256)
+    @NotNull
+    private String ordId;
+
+    @EdmIgnore
+    @Column(name = "app_id", length = 256)
+    @Convert("uuidConverter")
+    @TypeConverter(name = "uuidConverter", dataType = Object.class, objectType = UUID.class)
+    private UUID appId;
+
+    @Column(name = "local_id", length = 256)
+    @NotNull
+    private String localId;
+
+    @ElementCollection
+    @CollectionTable(name = "correlation_ids_entity_types", joinColumns = @JoinColumn(name = "entity_type_id", referencedColumnName = "id"))
+    private List<ArrayElement> correlationIds;
+
+    @Column(name = "level", length = 256)
+    @NotNull
+    private String level;
+
+    @Column(name = "title", length = 256)
+    @NotNull
+    private String title;
+
+    @Column(name = "short_description", length = 256)
+    @NotNull
+    private String shortDescription;
+
+    @Column(name = "description", length = Integer.MAX_VALUE)
+    private String description;
+
+    @Column(name = "system_instance_aware")
+    private boolean systemInstanceAware;
+
+    @ElementCollection
+    @CollectionTable(name = "changelog_entries_entity_types", joinColumns = @JoinColumn(name = "entity_type_id"))
+    private List<ChangelogEntry> changelogEntries;
+
+    @Column(name = "package_id")
+    @Convert("uuidConverter")
+    @TypeConverter(name = "uuidConverter", dataType = Object.class, objectType = UUID.class)
+    @NotNull
+    private UUID partOfPackage;
+
+    @EdmProtectedBy(name = "visibility_scope")
+    @Column(name = "visibility")
+    private String visibility;
+
+    @ElementCollection
+    @CollectionTable(name = "links_entity_types", joinColumns = @JoinColumn(name = "entity_type_id"))
+    private List<Link> links;
+
+    @ManyToMany(fetch = FetchType.LAZY)
+    @JoinTable(
+            name = "entity_type_product",
+            joinColumns = {@JoinColumn(name = "entity_type_id")},
+            inverseJoinColumns = {@JoinColumn(name = "product_id")})
+    private Set<ProductEntity> products;
```

Comment:
in the `ProductEntity` file add:

`    @ManyToMany(mappedBy = "products", fetch = FetchType.LAZY)
    private Set<EntityTypeEntity> entityTypes;`